### ISSUE TITLE
Help with read grouping + UMI error correction 

### ISSUE THREAD

**Issue by jindalk** [gui11aume/starcode#47]
Hi,
I have sequencing data where each read is divided into two parts: first 6 bases are `UMI` and the next 16 are the `read sequence`. I want to group sequences by `read sequence` and then within each group, perform UMI error correction. I'm using starcode-umi in the following manner:

 `./starcode-umi --umi-len 6 --umi-cluster s --umi-threads 4 --umi-d 1 --seq-d 0 u1.reads` 

but it seems like it isn't performing the function I intended. Am I doing something wrong? Is there a way to make starcode perform grouping followed by error correction?

Thanks

**Comment by gui11aume** [comment]
Hi @jindalk!

> but it seems like it isn't performing the function I intended.

Can you be more specific? What do you obtain and why is it not what you intend? Can you show a piece of the output that would help me understand what the problem is?

**Comment by jindalk** [comment]
Absolutely! Here is a snippet of my input:
<img width="188" alt="image" src="https://github.com/gui11aume/starcode/assets/46723936/6113ded2-9128-4533-880d-60b4626a367d">


Here is a snippet of my output:
<img width="215" alt="image" src="https://github.com/gui11aume/starcode/assets/46723936/e9cccdc1-5dc6-46c1-ab89-6e8cc8aa7f38">

Now when I load both my original reads and corrected reads in python and plot a venn diagram, I see that the `corrected reads` have several reads not found in the original data. I believe this shouldn't be the case during error correction?
<img width="462" alt="image" src="https://github.com/gui11aume/starcode/assets/46723936/84d16c75-144b-4500-806b-98f8adbc771b">

When I separately check the UMIs and Read seqs, there are no new UMIs and Read seqs in the corrected data, is my code somehow creating new UMI-Read seq pairs?

Happy to send over some files to reproduce this error.

Thanks

**Comment by gui11aume** [comment]
Thanks for clarifying the issue. I think you phrase it accurately:
> is my code somehow creating new UMI-Read seq pairs?

`starcode-umi` splits the reads, performs two independent rounds of clustering on each part and stitches them back together. In general, the process may create new UMI–read pairs.

From the description of your issue, I gather that you would like to first group the reads by read sequence, then run `starcode` on the UMIs of each read to find out how many (or which) UMIs are associated with each read. Let me know if I got it right.

We do not have an option to do this. One way to do it is to create an input file per read and automate the process of submitting it as an input to `starcode` (_e.g._,with `make` or with a for-loop). There is a bit of scripting work to generate the files in the first place, and then to gather the outputs into a single file.

This would be a useful feature, but we stopped developing `starcode` a long time ago (all the creators now have a different job) so it's unlikely that we will add it. The best option is to work with the available options and pass different inputs to `starcode`. I hope it still helps you solve your problem.
